pkg/boot/bls/bls.go

```diff
@@ -214,7 +214,7 @@ func parseLinuxImage(vals map[string]string, fsRoot string, variables map[string
 					if value == "" {
 						// If it's not found, fallback to look for default_kernelopts
 						log.Printf("kernelopts is empty, look for default_kernelopts\n")
-						if value, err = getGrubvalue(variables, "default_kernelopts"); value == "" {
+						if value, _ = getGrubvalue(variables, "default_kernelopts"); value == "" {
```

Comment:
I wonder if we should not print the value of err here. It might be useful when things go wrong. What if err is non-nil due to a disk IO error of some sort?

Reply:
I think we can ignore the error here, because we calling getGrubvalue twice, the only error it can return is `fmt.Errorf("variables map is nil")` is map is nil, so second invocation may return value or empty string:

```go
func getGrubvalue(variables map[string]string, key string) (string, error) {
	if variables == nil {
		// Only return error for nil variables map.
		return "", fmt.Errorf("variables map is nil")
	}
	if val, ok := variables[key]; ok && len(val) > 0 {
		return val, nil
	}
	return "", nil
}
```